src/hotspot/cpu/riscv/c1_Runtime1_riscv.cpp

```diff
@@ -332,9 +332,9 @@ static void restore_live_registers_except_r10(StubAssembler* sasm, bool restore_
     __ addi(sp, sp, 32 * wordSize);
   }
 
-  // if the number of popped regs is odd, zr will be added
-  // integer registers except ra(x1) & sp(x2) & x10
-  __ pop_reg(RegSet::range(x3, x9), sp);   // pop zr, x3 ~ x9
+  // if the number of popped regs is odd, the reserved slot for alignment will be removed
+  // integer registers except ra(x1) & sp(x2) & gp(x3) & tp(x4) & x10
+  __ pop_reg(RegSet::range(x5, x9), sp);   // pop zr, x5 ~ x9
```

Comment:
Is `zr` will be poped?

Reply:
Currently it seems not: it looks like [JDK-8278337](https://github.com/openjdk/riscv-port/pull/25) only removes the zr saving and restoring logic, but just remains that stack slot originally for zr unchanged.

Reply:
comment should be updated too though.

Reply:
Thanks for pointing it out -- seems there are unfixed comments at large out of my sight. I have fixed them according to the assembly code.